docs/templates.md

```diff
@@ -50,6 +50,62 @@ templates = Jinja2Templates(directory='templates')
 templates.env.filters['marked'] = marked_filter
 ```
 
+## Context processors
+
+A context processor is a function that returns a dictionary to be merged into a template context.
+Every function takes only one argument `request` and must return a dictionary to add to the context.
+
+A common use case of template processors is to extend the template context with shared variables.
+
+```python
+import typing
+from starlette.requests import Request
+
+def app_context(request: Request) -> typing.Dict[str, typing.Any]:
+    return {
+        'app': request.app,
+    }
+```
+
+### Registering context templates
+
+Pass context processors to `context_processors` argument of the `Jinja2Templates` class.
+
+```python
+import typing
+
+from starlette.requests import Request
+from starlette.templating import Jinja2Templates
+
+def app_context(request: Request) -> typing.Dict[str, typing.Any]:
+    return {'app': request.app}
+
+templates = Jinja2Templates(directory='templates', context_processors=[
+    app_context,
+])
+```
+
+### Asynchronous context processors
```

Comment:
I'm thinking if we should just avoid this section...

Reply:
Just a note about unsupported stuff. Will it work? Removing it completely is not helpful IMO.

Reply:
Yes. I'm fine with a simple note. :+1:

Reply:
done